### ISSUE TITLE
Proposal: Update ACT Rule methodologies to evaluate Computed Accessibility Tree (AOM) states instead of static CSS string parsing

### ISSUE THREAD

**Issue by graphic-designer** [w3c/wcag-act-rules#418]
### Context & Problem Statement
Currently, automated accessibility testing rules (ACT Rules) and compliance checkers evaluate text-resizing and layout-reflow capabilities by parsing static CSS stylesheets or checking for hardcoded absolute units (px). 

With the widespread implementation of Modern CSS (Baseline 2026), including container queries (@container style) and responsive layout methodologies, evaluating compliance through static string parsing of CSS causes severe architectural friction. Automated tools frequently trigger false positives ("Failures") on perfectly accessible interfaces simply because the validation engines evaluate the static code code-base rather than the actual rendered output.

### The Empirical Breakdown: Static vs. Runtime Evaluation
To provide immediate empirical proof of why static syntax parsing fails the modern web, consider the following live testing ecosystem built using current cross-browser Baseline features:

1. **The Live Interface (Baseline Reference):** https://graphic-designer.github.io/baseline-2026-mayo/index.html
2. **The Runtime Reality (100% Accessible):** [Google PageSpeed Runtime Audit](https://pagespeed.web.dev/analysis/https-graphic-designer-github-io-baseline-2026-mayo-index-html/uzu7uz36bz?form_factor=desktop) — When evaluated dynamically at the computed DOM/AOM level by a headless modern browser engine, the interface is completely flawless and accessible.
3. **The Static Failure (False Flags):** [W3C Jigsaw Validation Report](https://jigsaw.w3.org/css-validator/validator?uri=https%3A%2F%2Fgraphic-designer.github.io%2Fbaseline-2026-mayo%2Findex.html&profile=css3svg&usermedium=all&warning=1&vextwarning=&lang=es) — The official validator flags syntax errors simply because it reads flat text strings instead of actual browser outcomes.

### Additional Live Laboratory Evidence & Interactive Components
To further demonstrate the technical difference under User Agent Font Scaling (e.g., setting the browser configuration to "Very Large" font size without using optical zoom), you can analyze this standalone environment:

* **Interoperability Laboratory:** https://graphic-designer.github.io/2026-interoperability-laboratory

### Interactive Layout Implementation
The interoperability laboratory includes an interactive toggle button that allows real-time switching between a rigid px architecture and a responsive rem layout. 

When the user activates the relative scaling mechanism via this component:
* The text containers instantly shift to a fluid architecture driven by "@container" queries.
* The browser dynamically recalculates layout bounding boxes based on the user's native root font size preferences.
* The layout automatically adapts and "breathes" to prevent text overflow, proving that the user's biological configuration is respected in the final viewport.

### Technical Validation Failure (Visual Evidence)
The following official W3C validation output demonstrates this exact breakdown. Valid, modern CSS layouts utilizing "@container" queries are flagged as errors, penalizing automated compliance chains:

<img width="2922" height="2250" alt="Image" src="https://github.com/user-attachments/assets/4d656e8b-5615-44b8-80d8-7ec85e833ff9" />

#### Case A: Rigid Layout (Triggers no syntax errors, but fails the user)
```css
/* Static syntax checkers mark this as valid, but it overrides user autonomy */
.text-container {
  font-size: 16px; 
  width: 300px; /* Content overflows and breaks layout when the user overrides font size */
}
```

#### Case B: Modern Resilient Layout (Triggers false positives in static tools, but passes in reality)
```css
/* Automated tools flag this as unrecognized syntax or dynamic hazard, yet it perfectly preserves visual accessibility */
@container style(--theme: responsive) {
  .text-container {
    font-size: 1rem;
    width: auto; /* Dynamically resizes and reflows according to the User Agent base root size */
  }
}
```

### Technical Argument
When a user overrides browser settings to configure a custom root font size (User Agent Font Scaling), a resilient layout built with relative units (rem/em) adapts dynamically. 

Equating uniform Browser Zoom with native Font Scaling limits the scope of automated testing:
1. **Static CSS Parsing is Obsolete:** Evaluating stylesheets ignores how the browser actually computes layout boundaries using modern APIs.
2. **The AOM Solution:** Accessibility compliance should be an outcome-based evaluation. The only source of truth for an automated rule should be the computed properties within the **Accessibility Object Model (AOM)** at runtime.

### Proposed Change to ACT Methodology
We propose that the ACT Task Force updates the automated test rules for text-handling and reflow to shift their acceptance criteria:
* **From:** Inspecting static CSS syntax, absolute/relative units, or specific media query thresholds.
* **To:** Verifying the actual computed states of text containers and bounding boxes within the DOM/AOM under simulated User Agent text-scale configurations.

With all due respect to the committee, telling developers that *"regulators are just misinterpreting the law"* does not solve the operational gridlock. The W3C must take governance over its own validation ecosystem. This architectural shift ensures that automated validation rules remain future-proof, support modern CSS layouts, and prevent legal/compliance tools from penalizing valid, accessible engineering practices in regulated markets.

Best regards,
Ignacio Llorente Fdez de la Reguera

---

Proposal: Evolution of Conformance Testing for Modern CSS Interoperability
 #649 https://github.com/w3c/wcag3/issues/649

**Comment by WilcoFiers** [comment]
@graphic-designer Would you mind closing this issue and re-opening it into https://github.com/act-rules/act-rules.github.io/issues

I'd add that this issue doesn't really make much sense. You've clearly let an AI do most of this work, and it doesn't seem to have understood ACT rules. ACT rules don't say how things should be tested. The rules describe if an element should pass, fail or is inapplicable. It doesn't say how to evaluate that. This is left up to testers (tools or humans). Some tools do static code analysis, others do not.

Furthermore, ACT rules do use the accessibility tree and computed CSS properties. There are only a few situations I can think of where rules used declared properties rather than computed properties, and that was done deliberately because the rule needed to test if a declared property had the `!important` descriptor. That isn't on computed properties.
